### DISCUSSION TITLE
why?

### DISCUSSION THREAD

**Discussion by michabbb** [ryangjchandler/laravel-json-settings#3] [Q&A] [answered]
pardon, what does this package do better than existing mechanics that laravel already provides? what's the benefit?

thanks

**Comment by ryangjchandler** [comment, marked as answer]
Laravel doesn't provide a JSON config adapter out of the box, only PHP. 

Using a JSON file means that editing can be made simple via a GUI & the data can be persisted back to disk instead of storing in the database. 

The retrieval of the file is cached by default, so file I/O overhead is minimal and rare unless the file is constantly updated. 

Having different settings in production and local environments is also tough. The idea is to add a Git integration so that settings could be persisted back to Git from a Forge or Ploi server.
